src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java

```diff
@@ -141,7 +150,7 @@ public SecretKey deriveKey(String algorithm,
 
         private final byte[] label;
 
-        private SecretSchedule(String label) {
+        SecretSchedule(String label) {
```

Comment:
Why are we opening the visibility on this enum?  As far as I can tell it is only ever referenced within the SSLSecretDerivation class and could remain private if it hasn't been causing any problems up to now.

Reply:
From the JLS:

Reply:
From the JLS §8.8.3:  https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.3

`
If no access modifier is specified for the constructor of an enum type, the constructor is private. 
'

It's kind of like adding 'public' to interfaces.  They're not necessary.

Does that alleviate your concern?

Reply:
Yes it does.  I forgot that enums are private when there's no visibility explicitly stated.  So given that the whole thing looks good to me.